src/viewer.js

```diff
@@ -27,31 +27,67 @@ Object.defineProperty(Viewer.prototype, 'model', {
   }
 });
 
-Object.defineProperty(Viewer.prototype, 'progress', {
+Object.defineProperty(Viewer.prototype, 'frame', {
   enumerable: true,
-  set: function (progress) {
-    progress = Math.max(progress, 0);
-    progress = Math.min(progress, this.model.frames.length);
-    this._.progress = progress;
-    this._.renderer.data(this.model.frameGeometry(progress));
-    return this.progress;
+  set: function (frame) {
+    frame = Math.max(frame, 0);
+    frame = Math.min(frame, this.model.lastFrame);
+    this._.frame = frame;
+    this._.renderer.data(this.model.frameGeometry(frame));
+    return frame;
   },
 
   get: function () {
-    return this._.progress;
+    return this._.frame;
   }
 });
 
-Viewer.prototype.play = function () {
+Object.defineProperty(Viewer.prototype, 'step', {
+  enumerable: true,
+  get: function () {
+    return this.model.stepFromFrame(this.frame);
+  }
+});
+
+Viewer.prototype.render = function () {
   this._.renderer.play();
 };
 
+Viewer.prototype.play = function (frame) {
+  if (typeof frame === 'undefined') {
+    frame = this.model.lastFrame;
+  }
+
+  var spf = 1000 / this.model.fps;
+  var start, raf;
+  var delta = frame < this.frame ? -1 : 1;
+
+  var render = function (timestamp) {
+    var currentFrame = this.frame;
+    if (!start) start = timestamp;
+
+    if (currentFrame !== frame) {
+      if (timestamp - start > spf) this.frame += delta;
+      raf = requestAnimationFrame(render);
+    } else {
+      cancelAnimationFrame(raf);
```

Comment:
I think it's also dangerous to make multiple render calls into requestAnimationFrame. When I was playing around with the example, there were cases in which the animation would stop before the stop had finished. This behavior may be related to having multiple render functions created over the lifetime of the viewer. Consider making a single render function for each viewer instance.